src/com/amazon/ion/impl/IonReaderContinuableTopLevelBinary.java

```diff
@@ -97,6 +97,11 @@ public SymbolTable pop_passed_symbol_table() {
             return null;
         }
         symbolTableLastTransferred = currentSymbolTable;
+        if (symbolTableLastTransferred.isLocalTable()) {
+            // This method is called when transferring the reader's symbol table to either a writer or an IonDatagram.
+            // Those cases require a mutable copy of the reader's symbol table.
+            return ((_Private_LocalSymbolTable) symbolTableLastTransferred).makeCopy();
```

Comment:
`makeCopy()` is a `synchronized` method. Is this in a hot path in the code? Are there any performance concerns with this? If there are and it's feasible, is it worth making a second, unsynchronized `copy` method?

Reply:
This class only deals with `_Private_LocalSymbolTable` implementations provided by `IonReaderContinuableApplicationBinary` (i.e., `LocalSymbolTableSnapshot`), which does not mark this method `synchronized`. It is `synchronized` in `LocalSymbolTable`; I did not change that.